src/index.js

```diff
@@ -0,0 +1,235 @@
+      log(`Migrating to version ${migration.version} finished`)
+    }
+
+    if (!isDryRun) await repoVersion.setVersion(path, toVersion || getLatestMigrationVersion(migrations))
+    log('All migrations successfully migrated ', toVersion !== undefined ? `to version ${toVersion}!` : 'to latest version!')
+  } finally {
+    if (!isDryRun && !ignoreLock) await lock.close()
+  }
+}
+
+exports.migrate = migrate
+
+/**
+ * Main function to execute backward migration (reversion).
+ * It acquire lock on the provided path before doing any migrations.
+ *
+ * Signature of the progress callback is: function(migrationObject: object, currentMigrationNumber: int, totalMigrationsCount: int)
+ *
+ * @param {string} path - Path to initialized (!) JS-IPFS repo
+ * @param {int} toVersion - Version to which the repo will be reverted.
+ * @param {Object} options - Options for the reversion
+ * @param {function?} options.onProgress - Callback which will be called after each reverted migration to report progress
+ * @param {object?} options.repoOptions - Options that are passed to migrations, that can use them to correctly construct datastore. Options are same like for IPFSRepo.
+ * @param {boolean?} options.isDryRun - Allows to simulate the execution of the reversion without any effects. Make sense to utilize onProgress with this argument.
+ * @param {boolean?} options.ignoreLock - Won't lock the repo for reverting the migrations. Use with caution.
+ * @param {array?} options.migrations - Array of migrations to migrate. If undefined, the bundled migrations are used. Mainly for testing purpose.
+ * @returns {Promise<void>}
+ */
+async function revert (path, toVersion, { ignoreLock = false, repoOptions, onProgress, isDryRun = false, migrations }) {
+  migrations = migrations || defaultMigrations
+
+  if (!path) {
+    throw new errors.RequiredParameterError('Path argument is required!')
+  }
+
+  if (!(await repoInit.isRepoInitialized(path))) {
+    throw new errors.NotInitializedRepoError(`Repo in path ${path} is not initialized!`)
+  }
+
+  if (!toVersion) {
+    throw new errors.RequiredParameterError('When reverting migrations, you have to specify to which version to revert!')
+  }
+
+  if (!Number.isInteger(toVersion) || toVersion <= 0) {
+    throw new errors.InvalidValueError('Version has to be positive integer!')
+  }
+
+  const currentVersion = await repoVersion.getVersion(path)
+  if (currentVersion === toVersion) {
+    log('Nothing to revert, skipping reverting.')
+    return
+  }
+
+  if (currentVersion < toVersion) {
+    log(`Current repo's version (${currentVersion}) is lower then toVersion (${toVersion}), nothing to revert.`)
+    return
+  }
+
+  let reversibility = verifyReversibility(migrations, currentVersion, toVersion)
+  if (!reversibility.reversible) {
+    throw new errors.NonReversibleMigrationError(`Migration version ${reversibility.version} is not possible to revert! Cancelling reversion.`)
```

Comment:
```suggestion
    throw new errors.NonReversibleMigrationError(`It is not possible to revert to version ${toVersion} because migration version ${reversibility.version} is not reversible. Cancelling reversion.`)
```